bench/get_expanded_links.rb

```diff
@@ -0,0 +1,97 @@
+# /usr/bin/env ruby
+
+require ::File.expand_path('../../config/environment', __FILE__)
+require 'benchmark'
+
+require 'stackprof'
+
+abort "Refusing to run outside of development" unless Rails.env.development?
+
+large_reverse = Link.find_by_sql(<<-SQL).first[:target_content_id]
+  SELECT target_content_id
+  FROM links
+  WHERE links.link_type = 'parent'
+  GROUP BY target_content_id
+  ORDER BY COUNT (*) DESC
+  LIMIT 1
+SQL
+
+large_forward = LinkSet.find_by_sql(<<-SQL).first[:content_id]
+  SELECT content_id
+  FROM link_sets
+  WHERE id IN (
+    SELECT link_set_id
+    FROM links
+    GROUP BY link_set_id
+    ORDER BY COUNT(*) DESC
+    LIMIT 1
+  )
+SQL
+
+no_links = Link.find_by_sql(<<-SQL).first[:content_id]
+  SELECT content_id
+  FROM link_sets
+  LEFT JOIN links
+  ON links.link_set_id = link_sets.id
+  OR links.target_content_id = link_sets.content_id
+  WHERE links.id IS NULL
+  LIMIT 1
+SQL
+
+single_link = Link.find_by_sql(<<-SQL).first[:content_id]
+  SELECT content_id
+  FROM link_sets
+  INNER JOIN links ON links.target_content_id = link_sets.content_id
+  GROUP BY content_id
+  HAVING COUNT(*) = 1
+  LIMIT 1
+SQL
+
+def get_content_id_and_locale(content_id)
+  [
+    content_id,
+    Translation.join_content_items(
+      Queries::GetLatest.call(
+        ContentItemFilter.new(
+          scope: ContentItem.where(content_id: content_id)
+        ).filter(state: 'published')
+      )
+    ).pluck(:locale).first
+  ]
+end
+
+benchmarks = {
+  'Many reverse dependencies' => get_content_id_and_locale(large_reverse),
+  'Many forward dependencies' => get_content_id_and_locale(large_forward),
+  'No dependencies' => get_content_id_and_locale(no_links),
+  'Single link each way' => get_content_id_and_locale(single_link),
+}
+
+#ActiveRecord::LogSubscriber::IGNORE_PAYLOAD_NAMES.delete("SCHEMA")
+#ActiveRecord::Base.logger = Logger.new(STDOUT)
```

Comment:
Sure, then can we activate them by passing in arguments or similar? Otherwise they will just be ignored, or deleted in the future by someone who thinks they are unnecessary.